src/dials/algorithms/indexing/stills_indexer.py

```diff
@@ -64,27 +64,19 @@ def plot_displacements(reflections, predictions, experiments):
 
 
 def e_refine(params, experiments, reflections, graph_verbose=False):
-    # Stills-specific parameters we always want
-    assert (
-        params.refinement.reflections.outlier.algorithm
-        in (
-            None,
-            "null",
-        )
-    ), (
-        "Cannot index, set refinement.reflections.outlier.algorithm=null"
-    )  # we do our own outlier rejection
-
     from dials.algorithms.refinement.refiner import RefinerFactory
 
+    outlier_algorithm = params.refinement.reflections.outlier.algorithm
+    if outlier_algorithm == "sauter_poon":
```

Comment:
I notice that this means with mcd we can now have rejection during refinement, good, however this, along with the new L696, seems to cause the final returned `refined_reflections` table to include these rejected reflections which have xyzcal.mm set to (0,0,0), which then results in some very big rmsds being reported in `dials.ssx_index` for those crystals where reflections were rejected in the refiner. I think we some extra flags or filtering needs to be done, but I haven't worked out exactly where yet, any suggestions?